ibodata/lateral_profile.py

```diff
@@ -23,48 +23,79 @@ def penumbra_left(self):
         b = math.fabs(self.x_at_y(level, True))
         return (math.fabs(a - b) / (a + b)) * 200.0
 
     def flatness_50(self):
+        """
+        In case of corrupted data returns nan
+        Returns value between points if max/min value occurs on border
+        """
         d = (self.penumbra_left() + self.penumbra_right()) / 2
-        if d == np.nan:
+        if d != d:
             return np.nan
         left = self.x_at_y(0.5) + 2.0 * d
         right = self.x_at_y(0.5, True) - 2.0 * d
 
-        p_max = np.max(self.y[np.logical_and(self.x <= right, self.x >= left)])
-        p_min = np.min(self.y[np.logical_and(self.x <= right, self.x >= left)])
+        p_max = np.max(
+            np.append(self.y[np.logical_and(self.x <= right, self.x >= left)], [self.y_at_x(right), self.y_at_x(left)]))
+        p_min = np.min(
+            np.append(self.y[np.logical_and(self.x <= right, self.x >= left)], [self.y_at_x(right), self.y_at_x(left)]))
 
         return ((p_max - p_min) / (p_max + p_min)) * 100.0
 
     def flatness_90(self):
+        """
+        In case of corrupted data returns nan
+        Returns value between points if max/min value occurs on border
+        """
         d = (self.penumbra_left() + self.penumbra_right()) / 2
-        if d == np.nan:
+        if d != d:
             return np.nan
         left = self.x_at_y(0.9) + d
         right = self.x_at_y(0.9, True) - d
 
-        p_max = np.max(self.y[np.logical_and(self.x <= right, self.x >= left)])
-        p_min = np.min(self.y[np.logical_and(self.x <= right, self.x >= left)])
+        p_max = np.max(
+            np.append(self.y[np.logical_and(self.x <= right, self.x >= left)], [self.y_at_x(right), self.y_at_x(left)]))
+        p_min = np.min(
+            np.append(self.y[np.logical_and(self.x <= right, self.x >= left)], [self.y_at_x(right), self.y_at_x(left)]))
 
         return ((p_max - p_min) / (p_max + p_min)) * 100.0
 
     def asymmetry(self):
+        """
+        In case of corrupted data returns nan
+        Add area between mid and the nearest points if there's no 0 value in self.x
+        """
         area_left = np.trapz(self.y[self.x <= 0], self.x[self.x <= 0])
         area_right = np.trapz(self.y[self.x >= 0], self.x[self.x >= 0])
 
-        return ((area_left - area_right) / (area_left + area_right)) * 100.0
+        if np.argwhere(self.x == 0).size == 0:
+            left_index_arr = np.argwhere(self.x < 0)
+            area_left += np.trapz(np.append(self.y[left_index_arr[left_index_arr.size-1]], self.y_at_x(0)),
+                                  np.append(self.x[left_index_arr[left_index_arr.size-1]], [.0]))
+
+            right_index_arr = np.argwhere(self.x > 0)
+            area_right += np.trapz(np.append(self.y_at_x(0), self.y[right_index_arr[0]]),
+                                  np.append([.0], self.x[right_index_arr[0]]))
+
+        return float(((area_left - area_right) / (area_left + area_right)) * 100.0)
```

Comment:
Why conversion to float is needed ?

Reply:
If we want to check returned value in tests, returned value has to be some type that has round method.  Previous version was returning DataSet instead of float. We can convert DataSet to float in LateralProfile or tests. I decided to do it in LateralProfile

Reply:
OK, so I propose to explain it in some comments in the code (code should be easy to read for future developers).

Does it mean that `area_left` and `area_right` are also numpy DataSets and it propagates to return value ? If yes, then please put this info in comments.
numpy documentation of trapz says it returns floats:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.trapz.html

You may also consider introducing another variable called result (of DataSet type) and do `return float(result)`